darts/dataprocessing/transformers/boxcox.py

```diff
@@ -0,0 +1,102 @@
+"""
+Box-Cox Transformer
+-------------------
+"""
+
+from typing import Optional, Union, Sequence
+from scipy.stats import boxcox_normmax, boxcox
+from scipy.special import inv_boxcox
+
+from darts.timeseries import TimeSeries
+from darts.dataprocessing import Validator
+from darts.dataprocessing.transformers import FittableDataTransformer, InvertibleDataTransformer
+from darts.logging import get_logger, raise_if
+
+logger = get_logger(__name__)
+
+
+class BoxCox(FittableDataTransformer[TimeSeries], InvertibleDataTransformer[TimeSeries]):
+
+    def __init__(self,
+                 name: str = "BoxCox",
+                 validators: Optional[Sequence[Validator]] = None):
+        """
+        Box-Cox data transformer.
+        See https://otexts.com/fpp2/transformations.html#mathematical-transformations for more information
+
+        Parameters
+        ----------
+        name
+            A specific name for the transformer
+        validators
+            Sequence of validators that will be called before transform()
+        """
+        super().__init__(name=name, validators=validators)
+        self._lmbda = None
+
+    def fit(self,
+            data: TimeSeries,
+            lmbda: Optional[Union[float, Sequence[float]]] = None,
+            optim_method='mle') -> 'BoxCox':
+        """
+        Sets the `lmbda` parameter value.
+
+        Parameters
+        ----------
+        data
+            The time series to fit on
+        lmbda
+            If None given, will automatically find an optimal value of lmbda (for each dimension
+            of the time series) using `scipy.stats.boxcox_normmax` with `method=optim_method`
+            If a single float is given, the same lmbda value will be used for all dimensions of the series.
+            Also allows to specify a different lmbda value for each dimension of the time series by passing
+            a sequence of values.
```

Comment:
Even though you talk about it above, I would also add a docstring entry for `optim_method`.

Reply:
Oops forgot it, I'll add one.